inc/zoo/AnyCallable.h

```diff
@@ -65,8 +66,60 @@ struct AnyCallable<TypeErasureProvider, R(Args...)>: TypeErasureProvider {
                 const_cast<AnyCallable &>(*this)
             );
     }
+
+    void swap(AnyCallable& other) noexcept {
+        TypeErasureProvider::swap(other);
+        std::swap(targetInvoker_, other.targetInvoker_);
+    }
+
+    explicit operator bool() const noexcept {
+        return !empty();
+    }
+
+    const std::type_info& target_type() const noexcept {
+        return empty() ? typeid(void) : this->type();
+    }
+
+    bool empty() const noexcept {
+        return targetInvoker_ == emptyInvoker_;
+    }
+
+    template< class T >
+    T* target() noexcept {
```

Comment:
the syntax for pointers and references, `*` and `&` bind to the identifier.  I am in the tiny minority of people who prefer "East pointers and references" over "West pointers and references".  To reduce noise I like declarations of multiple values of the same type, there East pointers and references are better.